opm/autodiff/WellDensitySegmented.hpp

```diff
@@ -47,14 +47,14 @@ namespace Opm
         /// \param[in] b_perf       inverse ('little b') formation volume factor, size NP, P values per perforation
         /// \param[in] rsmax_perf   saturation point for rs (gas in oil) at each perforation, size N
         /// \param[in] rvmax_perf   saturation point for rv (oil in gas) at each perforation, size N
-        /// \param[in] surf_dens    surface densities for active components, size P
+        /// \param[in] surf_dens    surface densities for active components, size NP, P values per perforation
         static std::vector<double> computeConnectionDensities(const Wells& wells,
                                                               const WellStateFullyImplicitBlackoil& wstate,
                                                               const PhaseUsage& phase_usage,
                                                               const std::vector<double>& b_perf,
                                                               const std::vector<double>& rsmax_perf,
                                                               const std::vector<double>& rvmax_perf,
-                                                              const std::vector<double>& surf_dens);
+                                                              const std::vector<double>& surf_dens_perf);
```

Comment:
The surface density for the solvent + gas mix depends on the mixture. i.e. it is not a constant for all cells.